src/main/java/seedu/address/model/person/CourseContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,44 @@
+package seedu.address.model.person;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Tests that a {@code Person}'s {@code Course} matches any of the keywords given.
+ */
+public class CourseContainsKeywordsPredicate implements Predicate<Person> {
+    private final List<String> keywords;
+
+    public CourseContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        return keywords.stream()
+                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(person.getCourse().value, keyword));
```

Comment:
Thinking about it, this is needlessly redundant as well. Isn't a course code always a single word anyway?